### DISCUSSION TITLE
Typescript LSP does not work when an element is first added but when we start to type a "field"

### DISCUSSION THREAD

**Discussion by helmutgranda** [LazyVim/LazyVim#4416] [General]
The system works great and makes recommendations when I try an existing keyword (aka prop, aka field), but what I would like to see is recommendations when I first enter an empty area.

For example say I have a component named Box and has two properties, height and width.

- Box
- - heigth
- - width

When I start working on my file and I type

`<Box h`

The suggestion appears with `height`, however if I just type

`<Box ` -- note the empty space

I would like to see two options `height and width`, however nothing happens.

Is there anyway to force this behavior?

**Comment by zodman** [comment]
![image](https://github.com/user-attachments/assets/971bebe4-7b6e-4026-a979-f368e4f5aa8b)
Something you missing do you install the the correct LSP ?

**Reply by helmutgranda** [reply]
In this case you are getting the expected behavior which is what I described earlier. What happens if stead of "i" you just type `'<div ` (notice the space and nothing else). Do you get suggestions?

In my case I don't get anything at all and I would like to get the options available for (in your case) "div".
